doc/source/dev/toolchain.rst

```diff
@@ -125,13 +125,13 @@ Currently, SciPy wheels are being built as follows:
  Platform          Azure Base Image [5]_     Compilers                    Comment
 ================  ========================  ===========================  ==============================
 Linux (nightly)    ``ubuntu-20.04``          GCC 6.5                      See ``azure-pipelines.yml``
-Linux (release)    ``ubuntu-22.04``          GCC 8                        Built in separate repo [6]_
-OSX                ``macOS-10.15``           LLVM 12.0.0                  Built in separate repo [6]_
-Windows            ``windows-2019``          Visual Studio 2019 (vc142)   Built in separate repo [6]_
+Linux (release)    ``ubuntu-22.04``          GCC 8                        ``cibuildwheel``
+OSX                ``macOS-10.15``           LLVM 12.0.0                  ``cibuildwheel``
```

Comment:
The `macosx_x86_64` wheels are made on GHA using the `macos-11.0` image. This uses clang-13 and gfortran 11.3.

>    C compiler for the host machine: cc (clang 13.0.0 "Apple clang version 13.0.0 (clang-1300.0.29.30)")
    C linker for the host machine: cc ld64 711
    C++ compiler for the host machine: c++ (clang 13.0.0 "Apple clang version 13.0.0 (clang-1300.0.29.30)")
    C++ linker for the host machine: c++ ld64 711
    Cython compiler for the host machine: cython (cython 0.29.33)
    Host machine cpu family: x86_64
    Host machine cpu: x86_64
    Program python found: YES (/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-fa5b3_uy/cp310-macosx_x86_64/build/venv/bin/python)
    Found pkg-config: /usr/local/bin/pkg-config (0.29.2)
    Compiler for C supports arguments -Wno-unused-but-set-variable: NO
    Compiler for C supports arguments -Wno-unused-function: YES
    Compiler for C supports arguments -Wno-conversion: YES
    Compiler for C supports arguments -Wno-misleading-indentation: YES
    Compiler for C supports arguments -Wno-incompatible-pointer-types: YES
    Library m found: YES
    Fortran compiler for the host machine: gfortran (gcc 11.3.0 "GNU Fortran (GCC) 11.3.0")


The `macosx_arm64` wheels are made on cirrus-ci using the ghcr.io/cirruslabs/macos-monterey-xcode:13.3.1 image. I.e. on macOS-12. clang-13.1.6 and gfortran-12.1.0:

>  compiler for the host machine: cc (clang 13.1.6 "Apple clang version 13.1.6 (clang-1316.0.21.2.3)")
C linker for the host machine: cc ld64 762
C++ compiler for the host machine: c++ (clang 13.1.6 "Apple clang version 13.1.6 (clang-1316.0.21.2.3)")
C++ linker for the host machine: c++ ld64 762
Cython compiler for the host machine: cython (cython 0.29.33)
Host machine cpu family: aarch64
Host machine cpu: aarch64
Compiler for C supports arguments -Wno-unused-but-set-variable: YES 
Compiler for C supports arguments -Wno-unused-function: YES 
Compiler for C supports arguments -Wno-conversion: YES 
Compiler for C supports arguments -Wno-misleading-indentation: YES 
Compiler for C supports arguments -Wno-incompatible-pointer-types: YES 
Library m found: YES
Fortran compiler for the host machine: gfortran (gcc 12.1.0 "GNU Fortran (GCC) 12.1.0")